src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/CCSPostcodesBeanTest.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.ons.ctp.integration.contactcentresvc;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.junit.Test;
+import org.springframework.test.util.ReflectionTestUtils;
+import uk.gov.ons.ctp.common.event.EventPublisher.Channel;
+import uk.gov.ons.ctp.integration.contactcentresvc.config.AppConfig;
+import uk.gov.ons.ctp.integration.contactcentresvc.config.CCSPostcodes;
+
+public class CCSPostcodesBeanTest {
+
+  @Test
+  public void testDefaultList_findWithSpaceInSearchPostcode() throws IOException {
+    CCSPostcodesBean postcodesBean = createPostcodesBeanWithNoFile("GW12 AAA");
+    assertTrue(postcodesBean.isInCCSPostcodes("GW12 AAA"));
+  }
+
+  @Test
+  public void testDefaultList_findWithNoSpaceInSearchPostcode() throws IOException {
+    CCSPostcodesBean postcodesBean = createPostcodesBeanWithNoFile("GW12 AAA");
+    assertTrue(postcodesBean.isInCCSPostcodes("GW12AAA"));
+  }
+
+  @Test
+  public void testDefaultList_dontFindUnknownPostcode() throws IOException {
+    CCSPostcodesBean postcodesBean = createPostcodesBeanWithNoFile("GW12 AAA");
+    assertFalse(postcodesBean.isInCCSPostcodes("PO11 5AB"));
+  }
+
+  @Test
+  public void testPostcodeFile_findWithSpaceInSearchPostcode() throws IOException {
+    CCSPostcodesBean postcodesBean = createPostcodesBeanPostcodeFile("PO11 5AB", "AB12 7NB");
+    assertTrue(postcodesBean.isInCCSPostcodes("AB12 7NB"));
+  }
+
+  @Test
+  public void testPostcodeFile_findWithNoSpaceInSearchPostcode() throws IOException {
+    CCSPostcodesBean postcodesBean = createPostcodesBeanPostcodeFile("PO11 5AB", "AB12 7NB");
+    assertTrue(postcodesBean.isInCCSPostcodes("PO115AB"));
+  }
+
+  @Test
+  public void testPostcodeFile_dontFindUnknownPostcode() throws IOException {
+    CCSPostcodesBean postcodesBean = createPostcodesBeanPostcodeFile("PO11 5AB", "AB12 7NB");
+    assertFalse(postcodesBean.isInCCSPostcodes("GW12 AAA"));
+  }
+
```

Comment:
opportunity for additional test where the postcodes loaded into the CCS bean don't have spaces?